### PR TITLE
Added first implementation of digits QC task

### PR DESCRIPTION
**Pull request by gvolpe79**

### CODE REVIEW

File: Modules/HMPID/include/HMPID/HmpidTaskDigits.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+class TH2F;
+class TProfile;
+// class TProfile2D;
+
+using namespace o2::quality_control::core;
+
+namespace o2::quality_control_modules::hmpid
+{
+
+/// \brief Example Quality Control DPL Task
```
Comment (Barthelemy) [inline Modules/HMPID/include/HMPID/HmpidTaskDigits.h:34]: I would advise to put proper description.

File: Modules/HMPID/src/HmpidTaskDigits.cxx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+#include "HMPID/HmpidTaskDigits.h"
+#include "DataFormatsHMP/Digit.h"
+#include "DataFormatsHMP/Trigger.h"
+
+namespace o2::quality_control_modules::hmpid
+{
+
+HmpidTaskDigits::~HmpidTaskDigits()
+{
+  if (hOccupancyAvg) {
```
Comment (Barthelemy) [inline Modules/HMPID/src/HmpidTaskDigits.cxx:42]: There is no need anymore to check if it is not a nullptr. `delete nullptr` is valid and is a noop.

File: Modules/HMPID/src/HmpidTaskDigits.cxx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    }
+  }
+}
+
+void HmpidTaskDigits::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Info) << "initialize HmpidTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+
+  // this is how to get access to custom parameters defined in the config file at qc.tasks.<task_name>.taskParameters
+  if (auto param = mCustomParameters.find("myOwnKey"); param != mCustomParameters.end()) {
```
Comment (Barthelemy) [inline Modules/HMPID/src/HmpidTaskDigits.cxx:61]: please remove the boiler plate code.

File: Modules/HMPID/src/HmpidTaskDigits.cxx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+  }
+
+  hOccupancyAvg = new TProfile("hOccupancyAvg", "Occupancy per DDL;;Occupancy (%)", 14, 0.5, 14.5);
+  hOccupancyAvg->Sumw2();
+  hOccupancyAvg->SetOption("P");
+  hOccupancyAvg->SetMinimum(0);
+  hOccupancyAvg->SetMarkerStyle(20);
+  hOccupancyAvg->SetMarkerColor(kBlack);
+  hOccupancyAvg->SetLineColor(kBlack);
+  for (Int_t iddl = 0; iddl < 14; iddl++)
```
Comment (Barthelemy) [inline Modules/HMPID/src/HmpidTaskDigits.cxx:72]: Please use curly brackets even with one-liners.

File: Modules/HMPID/src/HmpidTaskDigits.cxx [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    hHMPIDchargeDist[i] = new TH1F(Form("hHMPIDchargeDist%i", i + 1), Form("Distribution of charges in chamber %i", i + 1), 2000, 0, 2000);
+    hHMPIDchargeDist[i]->SetXTitle("Charge (ADC)");
+    hHMPIDchargeDist[i]->SetYTitle("Entries/1 ADC");
+
+    hHMPIDdigitmapAvg[i] = new TH2F(Form("hHMPIDdigitmapAvg%i", i + 1), Form("Coordinates of hits in chamber %i", i + 1), 160, 0, 160, 144, 0, 144);
+    hHMPIDdigitmapAvg[i]->SetXTitle("padX");
+    hHMPIDdigitmapAvg[i]->SetYTitle("padY");
+
+    getObjectsManager()->startPublishing(hHMPIDchargeDist[i]);
+    getObjectsManager()->addMetadata(hHMPIDchargeDist[i]->GetName(), "custom", "34");
```
Comment (Barthelemy) [inline Modules/HMPID/src/HmpidTaskDigits.cxx:92]: is that really needed ?

File: Modules/HMPID/src/readout.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      },
+      "Activity": {
+        "number": "42",
+        "type": "2"
+      },
+      "monitoring": {
+	    "url": "infologger:///debug?qc"
+      },
+      "consul": {
+        "url": "http://consul-test.cern.ch:8500"
```
Comment (Barthelemy) [inline Modules/HMPID/src/readout.json:19]: Please set the url of consul to "".

**Review by Barthelemy — CHANGES_REQUESTED** [review]
This is good. Please consider the few comments I have written.

File: Modules/HMPID/src/HmpidTaskDigits.cxx (hunk 5 above)
Comment (gvolpe79) [reply Modules/HMPID/src/HmpidTaskDigits.cxx:92]: Do you refer to lines 91 and 92? I assume yes, but you are the expert, so if they are not needed I will remove them. I thought they are needed for the histograms publishing.

Comment (Barthelemy) [reply Modules/HMPID/src/HmpidTaskDigits.cxx]: I referred to 92
It adds metadata to a published object.

**Review by Barthelemy — APPROVED** [review]

**Pull request merged**